elastalert/config.py

```diff
@@ -236,6 +236,10 @@ def _dt_to_ts_with_format(dt):
         rule['compound_aggregation_key'] = rule['aggregation_key']
         rule['aggregation_key'] = ','.join(rule['aggregation_key'])
 
+    if isinstance(rule.get('compare_key'), list):
+        rule['compound_compare_key'] = rule['compare_key']
+        rule['compare_key'] = ','.join(rule['compare_key'])
```

Comment:
I'd rather we use ``compound_compare_key`` and not split a string every time. Let's set compound_compare_key to [rule['compare_key']] if compare_key is not a list.